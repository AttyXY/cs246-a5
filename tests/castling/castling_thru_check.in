game human human
move e2 e4
move e7 e5
move d2 d4
move d7 d5
move b1 c3
move b8 c6
move c1 g5
move c8 g4
move d1 d2
move d8 d7
move e1 c1
move e8 c8
move f2 f3
move f7 f6
move e1 c1
move e8 c8
quit
quit
