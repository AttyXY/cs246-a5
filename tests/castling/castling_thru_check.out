WELCOME TO CHESS! 
PLEASE ENTER ONE OF THE FOLLOWING COMMANDS TO BEGIN:
- setup
- game [player1] [player2]
- quit (to exit)

STARTING GAME!

8  rnbqkbnr
7  pppppppp
6  --------
5  --------
4  --------
3  --------
2  PPPPPPPP
1  RNBQKBNR
   abcdefgh

8  rnbqkbnr
7  pppppppp
6  --------
5  --------
4  ----P---
3  --------
2  PPPP-PPP
1  RNBQKBNR
   abcdefgh

8  rnbqkbnr
7  pppp-ppp
6  --------
5  ----p---
4  ----P---
3  --------
2  PPPP-PPP
1  RNBQKBNR
   abcdefgh

8  rnbqkbnr
7  pppp-ppp
6  --------
5  ----p---
4  ---PP---
3  --------
2  PPP--PPP
1  RNBQKBNR
   abcdefgh

8  rnbqkbnr
7  ppp--ppp
6  --------
5  ---pp---
4  ---PP---
3  --------
2  PPP--PPP
1  RNBQKBNR
   abcdefgh

8  rnbqkbnr
7  ppp--ppp
6  --------
5  ---pp---
4  ---PP---
3  --N-----
2  PPP--PPP
1  R-BQKBNR
   abcdefgh

8  r-bqkbnr
7  ppp--ppp
6  --n-----
5  ---pp---
4  ---PP---
3  --N-----
2  PPP--PPP
1  R-BQKBNR
   abcdefgh

8  r-bqkbnr
7  ppp--ppp
6  --n-----
5  ---pp-B-
4  ---PP---
3  --N-----
2  PPP--PPP
1  R--QKBNR
   abcdefgh

8  r--qkbnr
7  ppp--ppp
6  --n-----
5  ---pp-B-
4  ---PP-b-
3  --N-----
2  PPP--PPP
1  R--QKBNR
   abcdefgh

8  r--qkbnr
7  ppp--ppp
6  --n-----
5  ---pp-B-
4  ---PP-b-
3  --N-----
2  PPPQ-PPP
1  R---KBNR
   abcdefgh

8  r---kbnr
7  pppq-ppp
6  --n-----
5  ---pp-B-
4  ---PP-b-
3  --N-----
2  PPPQ-PPP
1  R---KBNR
   abcdefgh

8  r---kbnr
7  pppq-ppp
6  --n-----
5  ---pp-B-
4  ---PP-b-
3  --N-----
2  PPPQ-PPP
1  R---KBNR
   abcdefgh

Invalid move. Input another move.
8  r---kbnr
7  pppq-ppp
6  --n-----
5  ---pp-B-
4  ---PP-b-
3  --N-----
2  PPPQ-PPP
1  R---KBNR
   abcdefgh

Invalid move. Input another move.
8  r---kbnr
7  pppq-ppp
6  --n-----
5  ---pp-B-
4  ---PP-b-
3  --N--P--
2  PPPQ--PP
1  R---KBNR
   abcdefgh

8  r---kbnr
7  pppq--pp
6  --n--p--
5  ---pp-B-
4  ---PP-b-
3  --N--P--
2  PPPQ--PP
1  R---KBNR
   abcdefgh

8  r---kbnr
7  pppq--pp
6  --n--p--
5  ---pp-B-
4  ---PP-b-
3  --N--P--
2  PPPQ--PP
1  --KR-BNR
   abcdefgh

8  --kr-bnr
7  pppq--pp
6  --n--p--
5  ---pp-B-
4  ---PP-b-
3  --N--P--
2  PPPQ--PP
1  --KR-BNR
   abcdefgh


PLEASE ENTER ONE OF THE FOLLOWING COMMANDS TO CONTINUE:
- setup (to create new setup)
- game (to continue with previous setup)
- quit (to exit)

THANKS FOR PLAYING CHESS!
Final Score:
White: 0
Black: 0
